fsm/src/main/java/com/teamdev/fsm/CharSequenceReader.java

```diff
@@ -29,20 +30,26 @@ public String readOperator(){
 
         StringBuilder operator = new StringBuilder();
 
-        if (read() == '>' || read() == '<'){
+        int startPosition = position();
+
+        while(canRead() && isOperator(read())){
             operator.append(read());
             incrementPosition();
-            if (read() == '='){
-                operator.append(read());
-            }
-            else decrementPosition();
-
         }
-        else operator.append(read());
+
+        if (startPosition != position())
+            decrementPosition();
 
         return operator.toString();
     }
 
+    public boolean isOperator(char sign){
```

Comment:
CharSequenceReader should know nothing about operators. Remove all the code concerning operators.